docs/sgp40/raspisetup.rst

```diff
@@ -0,0 +1,96 @@
+
++-----------+------------------+
+| *SGP40*   | *Raspberry Pi*   |
++===========+==================+
+| VCC       | Pin 1            |
++-----------+------------------+
+| GND       | Pin 6            |
++-----------+------------------+
+| SDA       | Pin 3            |
++-----------+------------------+
+| SCL       | Pin 5            |
++-----------+------------------+
+
+|RaspberryPi Pinout|
+
+Raspberry Pi
+~~~~~~~~~~~~
+
+-  **Install the newest Raspberry Pi OS on to your Raspberry Pi. If you
+   don't know how to do that, there is a guide on the Raspberry Pi
+   website**
+   (https://projects.raspberrypi.org/en/projects/raspberry-pi-setting-up).
+-  Enable the I2C interface in the Raspberry Pi config.
+    -  Open a terminal.
+    -  Run ``sudo raspi-config``.
+    -  Select "5 Interfacing Options Configure connections to peripherals".
+    -  Select "P5 I2C Enable/Disable automatic loading of I2C kernel module".
+    -  Select "Yes" when questioned "Would you like the ARM I2C interface to be enabled?".
+    -  The Raspberry Pi should respond with "The ARM I2C interface is enabled". Confirm with "Ok"
+-  Download driver.
+    -  Go to the Sensirion SGP Driver Release page (https://github.com/Sensirion/embedded-sgp/releases) and download the ``sgp40-VERSION.zip`` file.
+    -  Unzip the file into the directory where you want to install the driver (Run: ``unzip sgp40-VERSION.zip`` ).
+-  Adapt the driver to work with Raspberry Pi.
+    -  We use the linux\_user\_space implementation of the hw\_i2c to run the driver on the Raspberry Pi. To use the in the project provided sample implementation navigate to the ``./hw_i2c/sample-implementations/linux_user_space`` direcory.
+    -  Copy the file ``sensirion_hw_i2c_implementation.c`` from there and replace the file named the same directly in ``./hw_i2c`` or just run this command in the root directory of your project:
+   ``cp ./hw_i2c/sample-implementations/linux_user_space/sensirion_hw_i2c_implementation.c ./hw_i2c/``.
+-  Compile the driver.
+    -  Run ``make`` in the root directory of your project.
+
+Output:
+
+::
+
+    rm -f sgp40_example_usage
+    cc -Os -Wall -fstrict-aliasing -Wstrict-aliasing=1 -Wsign-conversion -fPIC -I. -I. -I. -I./hw_i2c -o sgp40_example_usage ./sensirion_arch_config.h ./sensirion_i2c.h ./sensirion_common.h ./sensirion_common.c ./sgp_git_version.h ./sgp_git_version.c ./sgp40.h ./sgp40.c ./hw_i2c/sensirion_hw_i2c_implementation.c ./sgp40_example_usage.c
+
+-  **To test the setup run the example code which was just compiled with**
+   ``./sgp40_example_usage`` **.**
+
+Output:
+
+::
+
+    SGP40 driver version 6.0.0-32-gc792a99
+    SGP sensor probing successful
+    SerialID: 00:00:02:37:BF:3D
+    sraw: 30403
+    sraw: 30414
+    sraw: 30407
+    sraw: 30403
+    sraw: 30390
+    sraw: 30379
+    sraw: 30370
+    sraw: 30383
+    ...
+
+Troubleshooting
+---------------
+
+SGP sensor probing failed
+~~~~~~~~~~~~~~~~~~~~~~~~~
```

Comment:
Maybe state the exact error: "SGP sensor probing failed" if I remember correctly.

Reply:
You don't ;)